vault/seal.go

```diff
@@ -110,6 +134,10 @@ func (d *defaultSeal) SetAccess(access *seal.Access) {
 
 func (d *defaultSeal) SetCore(core *Core) {
 	d.core = core
+	if d.logger == nil {
+		d.logger = d.core.Logger().Named("defaultseal")
```

Comment:
We could give this a better name (e.g. shamir-seal or autoseal-recovery) by looking at unsealKeyPath, right?